api/src/modules/admin-regions/admin-region.repository.ts

```diff
@@ -91,10 +108,21 @@ export class AdminRegionRepository extends ExtendedTreeRepository<
      * In this case we can get an Intersection with the radius created by this sourcing location so we get the highest level
      * as doing this is more performant that getting the geometry and intersecting with it
      */
-
-    return res.reduce(function (previous: any, current: any) {
+    const result: any = res.reduce(function (previous: any, current: any) {
       return previous.level > current.level ? previous : current;
     });
+    if (country) {
+      const adminRegion: AdminRegion = await this.findOneOrFail({
+        id: result.adminRegionId,
+      });
+      const ancestor: AdminRegion = await this.findAncestorsTree(adminRegion);
```

Comment:
Try if this works and avoid 2 roundtrips to DB
```suggestion
      const ancestor: AdminRegion = await this.findAncestorsTree({id: result.adminRegionId} as AdminRegion);
```